src/TropicalGeometry/semiring_map.jl

```diff
@@ -1,23 +1,78 @@
 ################################################################################
 #
-#  Tropical semiring maps
-#  ======================
-#  maps from a field K to a tropical semiring T with the purpose of encoding
-#    - a valuation on K
-#    - a choice of min- or max-convention
+#  Tropical semiring map
 #
 ################################################################################
-struct TropicalSemiringMap{typeofValuedField,typeofUniformizer,minOrMax}
-    valued_field::typeofValuedField
+
+@doc raw"""
+    TropicalSemiringMap{DomainType, UniformizerType, MinOrMax}
+
+A structure representing a map from a valued field or ring to a tropical
+semiring, encoding a valuation and a choice of min- or max-convention.
+
+# Parameters
+- `DomainType`: type of valued field (if specified)
+    or type of valued ring (if valued field unspecified)
+- `UniformizerType`: type of uniformizer in valued ring
+- `MinOrMax`: either `typeof(min)` or `typeof(max)` depending on convention
+
+# Fields
+- `valued_field`: the valued field. can be `nothing` if valued field unspecified
+- `uniformizer_field`: the uniformizer as element in the valued field
+    can be `nothing` if valued field unspecified or valuation trivial
+- `valued_ring`: the valued ring, either specified by user or
+    inferred from the valued field specified by user
+- `uniformizer_ring`: the uniformizer as element in the valued ring specified by user
+    can be `nothing` if valuation trivial
+- `residue_field`: the residue field, inferred from fields above
+- `tropical_semiring`: either `tropical_semiring(min)` or `tropical_semiring(max)`
+    depending on convention
+"""
+struct TropicalSemiringMap{DomainType, UniformizerType, MinOrMax}
+    valued_field::Union{Nothing,Field}
     uniformizer_field::Union{Nothing,FieldElem}
     valued_ring::Ring
-    uniformizer_ring::typeofUniformizer
+    uniformizer_ring::Union{Nothing,RingElem}
```

Comment:
Off-topic comment: the names `uniformizer_field` and `uniformizer_ring` are unfortunate as I'd expect those to be either a field resp. ring, or functions returning such.

Of course that's existing code, so I am not expecting this to be changed in this PR, but it might be worth contemplating for future refactoring whether better names can be found (`uniformizer_in_ring` and `uniformizer_in_field` perhaps?)